policy/query-builder/query-builder.go

```diff
@@ -0,0 +1,214 @@
+package query_builder
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"math/rand"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/wopta/goworkspace/lib"
+)
+
+var (
+	paramsHierarchy = []map[string][]string{
+		{"codeCompany": []string{"codeCompany"}},
+
+		{"proposalNumber": []string{"proposalNumber"}},
+
+		{"insuredFiscalCode": []string{"insuredFiscalCode"}},
+
+		{"contractorName": []string{"contractorName", "contractorSurname"}},
+		{"contractorSurname": []string{"contractorName", "contractorSurname"}},
+
+		{"startDateFrom": []string{"startDateFrom", "startDateTo", "company", "product", "producerUid", "reserved", "status", "payment", "renewMonth"}},
+		{"startDateTo": []string{"startDateFrom", "startDateTo", "company", "product", "producerUid", "reserved", "status", "payment", "renewMonth"}},
+		{"company": []string{"startDateFrom", "startDateTo", "company", "product", "producerUid", "reserved", "status", "payment", "renewMonth"}},
+		{"product": []string{"startDateFrom", "startDateTo", "company", "product", "producerUid", "reserved", "status", "payment", "renewMonth"}},
+		{"producerUid": []string{"startDateFrom", "startDateTo", "company", "product", "producerUid", "reserved", "status", "payment", "renewMonth"}},
+		{"status": []string{"startDateFrom", "startDateTo", "company", "product", "producerUid", "reserved", "status", "payment", "renewMonth"}},
+		{"payment": []string{"startDateFrom", "startDateTo", "company", "product", "producerUid", "reserved", "status", "payment", "renewMonth"}},
+		{"renewMonth": []string{"startDateFrom", "startDateTo", "company", "product", "producerUid", "reserved", "status", "payment", "renewMonth"}},
```

Comment:
it seems to be missing the following entry:
```
{"reserved": []string{....}}
```

otherwise it should be removed from the allowedLists